### ISSUE TITLE
Successful start but no device on guest VM present

### ISSUE THREAD

**Issue by JimKnopfIoT** [tdhoward/COMpipe#15]
I plugged my device which is detected on the Win 11 Pro hostsystem as COM4. I started the Win10 VM and run in Powershell on the hostsystem the commands:

> Set-VMComPort -VMName 'Win10' -Number 1 -Path \\.\pipe\MyDevice

> COMpipe.exe -c \\.\COM4 -p \\.\pipe\MyDevice
Serial port opened successfully.
Named pipe opened successfully.
Press Q to quit.

No errors. But on my guest system i can't use the device. I tested different types of devices. I used the precompiled COMpipe.exe.
Any hint where to look at?

**Comment by tdhoward** [comment]
Hmm.  Does COM1 show up in your device manager on the WIN10 VM?  If so, you could try connecting to it with a simple terminal program to see if anything is coming through.
Also, I think it's usually "\\.\pipe\...", not "\.\pipe\..." but I don't know if that would make a difference.  The fact that the pipe exists and is opened successfully is a good sign, though.

**Comment by RUfreestyler** [comment]
I have the same problem on virtual machine with Windows 7, device is not available on VM.

**Comment by JimKnopfIoT** [comment]
Right now i try it using a SH-U09C5 USB to TTL Adapter. The adapter works on the Windows 11 hostsystem on COM4. I can connect on the hostsystem to a target via UART. 

When i use COMpipe.exe -c \.\COM4 -p \.\pipe\MyDevice 
it tells me:
Serial port opened successfully.
Named pipe opened successfully.
Press Q to quit.
In my Windows 10 VM there are 2 COM Ports. COM1 should work with the SH-09C5 USB to TTL Adapter.
I use Putty on the hostsystem and also in the VM. In the VM putty open up a window but i get only a black screen. There is no connection to the target.

The SH-U09C5 USB to TTL Adapter has two LED to indicate TX and RX traffic. When i use it on the hostsystem, both LEDs are flashing when i enter commands in putty (connected to a target device).

When i do the same in the VM with successfully established COMpipe connection, only the TX LED is flashing.

**Comment by JimKnopfIoT** [comment]
I tried to add a different baudrate and that fixed it with the SH-U09C5 USB to TTL Adapter.
PS C:\Users\sw> COMpipe.exe -b 115200 -c \\.\COM4 -p \\.\pipe\MyDevice
Serial port opened successfully.
Named pipe opened successfully.
Press Q to quit.
Now putty works on VM guest with RX/TX LED both flashing when i enter commands to the target device (connected via UART).
![vmconnect_MUtMYMyVnF](https://github.com/tdhoward/COMpipe/assets/48170499/44d1df14-d905-4b79-8111-83e923462790)

I will check this with my PL2303GC device too.

**Comment by JimKnopfIoT** [comment]
The PL2303GC device is shown in Windows 11 hostsystem and it uses 38400 baud.
![mmc_ujzwMoTXPt](https://github.com/tdhoward/COMpipe/assets/48170499/c16a8e9c-151b-4aaf-a548-f991d4eb7ca6)

This device does not exist in the guest VM. I installed the drivers for this device in the guest VM but i can't connect to it.

**Comment by JimKnopfIoT** [comment]
I don't know why this PL2303GC device has issues. I tested other devices like a JTAGulator, BusPirate or TIAO Tumpa without problems. At the moment, only this PL2303GC device will not work. 
![Uploading vmconnect_0rxHhHCvUR.png…]()

**Comment by JimKnopfIoT** [comment]
I measured the frequeny on the PL2303GC IC (Pin 5, RX) when connecting the device to a working VM on my Linux Laptop (KVM). 7,75µs, means 129.032 baud or round about 129.000 baud which is very uncommon. 8,67µs would be 115.200 baud which is a default value. I tried it with different special settings with no luck.
![Data-RX_Connect-device_timing_high](https://github.com/tdhoward/COMpipe/assets/48170499/9dba8708-05a7-4c5a-8aab-c00ca858bdef)
![WindowsTerminal_PFqqywR2jd](https://github.com/tdhoward/COMpipe/assets/48170499/1836a8d7-a1e7-4e7c-8e01-07f8a58ae423)

**Comment by tdhoward** [comment]
Thanks for your work and comments on this @JimKnopfIoT .  Generally speaking, you won't need any drivers on the VM, since it has no access to the hardware, and COMpipe only moves the data across without regard for hardware signals like CTS, etc. Maybe I should put in some code to allow for very flexible baud rates, since that seems to be a potential problem.

**Comment by JimKnopfIoT** [comment]
@tdhoward I don't think it's the baudrate causing problems. I tried a commercial  product for usb passtrough (trial version). It worked like a charm with 38.400 baud default setting with this commercial product. My experiemnts with this PL2303GC ends here. I found out what i wanted to know. So it's ok if COMpipe doesn't do the job for this specific IC/device while all my other devices are working fine now. Maybe other users have same experiences with this IC. From my point, the issue can be closed.
